### DISCUSSION TITLE
error: expected auto, relative length, fraction, integer, or array, found content

### DISCUSSION THREAD

**Discussion by psads-git** [typst/typst#4509] [Q&A] [answered]
I am getting the following error with the code below. Could you please help me? Thanks!

```
error: expected auto, relative length, fraction, integer, or array, found content
   ┌─ /tmp/x.typ:61:18
   │
61 │             rows: size,
   │                   ^^^^
```

```
#let calsize(body) = context {
  let size = measure(body)
  return size.height
}

#let txt = "Hello"

#let size = calsize([#txt])

#block(
  breakable: false,
  grid(
    columns: 2,
    rows: size,
    gutter: 0.5em,
    enum.item(1)[#txt],
    enum.item(2)[#txt],
    enum.item(3)[#txt],
    enum.item(4)[#txt],
    enum.item(5)[#txt])
)

```

**Comment by TS60** [comment]
Here is the minified problem:

```typst
#let calsize(body) = context {
  let size = measure(body)
  assert(type(size.height) == length)
  return size.height
}

#let txt = "Hello"

#let size = calsize([#txt])

#type(size) // is now content instead of length
```

Either a typst compiler bug or maybe a misuse of `context`.

**Reply by laurmaedje** [reply]
Working as intended. The context block turns anything into content. Everything depending on the contextual values must be inside. This also is the case through function call boundaries.

**Reply by psads-git** [reply]
Thanks, @TS60 and @laurmaedje, for your replies.

Is there some way of converting content to length? That would be a workaround.

**Reply by laurmaedje** [reply, marked as answer]
Sorry, I was on my phone earlier, otherwise I'd have sent a working snippet. Here it is:
```typ
#let calsize(body) =  {
  let size = measure(body)
  return size.height
}

#let txt = "Hello"

#context block(
  breakable: false,
  grid(
    columns: 2,
    rows: calsize([#txt]),
    gutter: 0.5em,
    enum.item(1)[#txt],
    enum.item(2)[#txt],
    enum.item(3)[#txt],
    enum.item(4)[#txt],
    enum.item(5)[#txt])
)
```

The `context` must really wrap everything that depends on it. Converting contextual content to length is not possible because the length really isn't known at this point.

**Comment by psads-git** [comment]
Thanks a lot, @laurmaedje, that works!
